### PR TITLE
add simple pollard dl solver

### PR DESCRIPTION
**Pull request by boneyard93501**
it's pretty slow. so this is a starting point :)

### CODE REVIEW

File: src/elgamal.rs [hunk 1]
```diff
@@ -215,6 +216,10 @@ impl ExponentElGamal {
         _sk: &ElGamalPrivateKey,
     ) -> Result<BigInt, ElGamalError> {
         //TODO
```
Comment (doronz2) [inline src/elgamal.rs]: This is not a traditional EG decrypt. Can you move it to the test module?

File: src/dl_solvers.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+        loop {
+            self.step(&mut x, &mut a, &mut b);
+            self.step(&mut big_x, &mut big_a, &mut big_b);
+            self.step(&mut big_x, &mut big_a, &mut big_b);
+            if x == big_x {
+                break;
+            }
+        }
+        x
```
Comment (doronz2) [inline src/dl_solvers.rs]: Shouldn't you  extract the exponent from a-big_a / b- big_a

Comment (doronz2) [reply src/dl_solvers.rs]: and also check that b != big_b

File: src/elgamal.rs (hunk 1 above)
Comment (boneyard93501) [reply src/elgamal.rs]: yep

File: src/dl_solvers.rs (hunk 2 above)
Comment (boneyard93501) [reply src/dl_solvers.rs]: since we're using an additive random walk, i think we're ok.
yep to check.

Comment (doronz2) [reply src/dl_solvers.rs]: I don't think that x is the right value. In the end of the loop you need to extract the exponent.

Comment (doronz2) [reply src/dl_solvers.rs]: Do you need help in fixing this? (all you got to do it to return the value  a-big_a / b- big_a)

**Comment by boneyard93501** [comment]
sorry for the lag and confusion, i kept looking at my local version i failed to push. what i got is
```rust

let nom = BigInt::mod_sub(&a, &big_a, &self.big_p);
let denom = BigInt::mod_sub(&big_b, &b, &self.big_p);
let res = BigInt::modulus(&(BigInt::mod_inv(&denom, &self.big_p) * &nom), &self.big_p);
Some(res)
```

**Comment by doronz2** [comment]
Perfect! Thanks!

**Pull request merged**